packages/components/src/mobile/dark-mode/index.native.js

```diff
@@ -0,0 +1,37 @@
+/**
+ * External dependencies
+ */
+import { eventEmitter, initialMode } from 'react-native-dark-mode';
+import React from 'react';
+
+export function useStyle( light, dark, theme ) {
+	const finalDark = {
+		...light,
+		...dark,
+	};
+
+	return theme === 'dark' ? finalDark : light;
+}
+
+// This function takes a component...
+export function withTheme( WrappedComponent ) {
+	return class extends React.Component {
+		constructor( props ) {
+			super( props );
+
+			this.state = {
+				mode: initialMode,
+			};
+		}
+
+		componentDidMount() {
+			eventEmitter.on( 'currentModeChanged', ( newMode ) => {
+				this.setState( { mode: newMode } );
```

Comment:
Personal preference, but if `mode` only has two valid states ('light' and 'dark'), it would be easier for me to follow this if the state had something like an `isLightTheme` boolean instead of a `mode` string. We could then pass the boolean to the wrapped component, and it would be obvious from the wrapped component that `isLightTheme` is a boolean variable with two possible states.